### ISSUE TITLE
Docker-compose ps on VIC displays incorrect data

### ISSUE THREAD

**Issue by sflxn** [vmware/vic#2508]
Docker-compose ps against VIC is displaying odd data.  First, it's displaying 2 of each.  Second, it's not displaying the first letter of a service.

Expected:

```
loc@ubuntu:~/docker-tests/compose/wordpress$ docker-compose ps
        Name                       Command               State          Ports         
-------------------------------------------------------------------------------------
wordpress_db_1          docker-entrypoint.sh mysqld      Up      3306/tcp             
wordpress_wordpress_1   /entrypoint.sh apache2-for ...   Up      0.0.0.0:8000->80/tcp 
```

Actual:

```
loc@ubuntu:~/docker-tests/compose/wordpress$ docker-compose ps
        Name                      Command               State          Ports         
------------------------------------------------------------------------------------
ordpress_db_1          docker-entrypoint.sh mysqld      Up      3306/tcp             
ordpress_db_1          docker-entrypoint.sh mysqld      Up      3306/tcp             
ordpress_wordpress_2   /entrypoint.sh apache2-for ...   Up      0.0.0.0:8000->80/tcp 
ordpress_wordpress_2   /entrypoint.sh apache2-for ...   Up      0.0.0.0:8000->80/tcp 
```

Here's the result of docker ps:

```
loc@ubuntu:~/docker-tests/compose/wordpress$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                      PORTS               NAMES
c06ef8c7d7c6        wordpress:latest    "/entrypoint.sh apach"   5 days ago          Exited (1) 13 minutes ago                       wordpress_wordpress_2
1c3a3f5a3735        mysql:5.7           "docker-entrypoint.sh"   5 days ago          Up 14 minutes                                   wordpress_db_1
```

**Comment by mdubya66** [comment]
Is this similar to https://github.com/vmware/vic/issues/2091

**Comment by chengwang86** [comment]
Update:

This is because we set the "scope" of "annotation" to "hidden" in lib/config/executor/container_vm.go

Changing it to "read-only" solves the problem. 

Another workaround is to copy the annotation/labels in the extraconfig manually when refreshing the container but keeping "annotation" hidden to the container.

Need to discuss with @hickeng about

**Comment by chengwang86** [comment]
Update: @sflxn 

When doing "docker-compose -H ... ps", it calls the function containers() in /usr/local/lib/python2.7/dis-packages/docker/api/container.py, instead of our docker-persona.

In the containers() function:
`
u = self._url("/containers/json")
res = self._result(self._get(u, params=params), True)
`
On vic, "res" returns containers that satisfy "params" with "Labels: None", whereas on regular docker it returns with "Labels: {"com.docker.compose.oneoff":'False' ...}".

Seems like the container labels are missing when docker-compose tries to obtain the container info from vic.

**Comment by sflxn** [comment]
@mdubya66 I think there are two issues in this one.  @chengwang86 can fill in the details, but we should narrow the scope of this issue to just the truncation of the names.  The duplicate name issue is due to the fact we do not have docker ps filtering #1676, and that one is iceboxes.

**Comment by chengwang86** [comment]
Update:

The duplicate container information is due to #1676 enable filter for docker ps
Docker-compose needs the following two filters to obtain information from all the desired containers:

- com.docker.compose.project=vic
- com.docker.compose.oneoff=False

"Docker-compose ps" would make two queries: (i) oneoff=False (ii) oneoff=True. Since "filter" is not implemented, "Docker-compose ps" would return the same container information twice as shown in this issue.

Example of "docker ps --filter":

Expected behavior:
```
docker ps -a -f label=com.docker.compose.oneoff=True
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
docker ps -a -f label=com.docker.compose.oneoff=False
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                    PORTS               NAMES
f0cc5d78fe0f        busybox             "sh"                16 hours ago        Exited (0) 16 hours ago                       vic_web_1
```

Actual behavior:
```
docker -H 192.168.60.148:2375 ps -a -f label=com.docker.compose.oneoff=True
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                           NAMES
11af9dddea63        busybox             "sh"                4 days ago          Up 10 hours         192.168.60.148:8080->8080/tcp   vic_web_1
docker -H 192.168.60.148:2375 ps -a -f label=com.docker.compose.oneoff=False
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                           NAMES
11af9dddea63        busybox             "sh"                4 days ago          Up 10 hours         192.168.60.148:8080->8080/tcp   vic_web_1
```

**Comment by chengwang86** [comment]
The truncated container name is fixed by #3188 

Duplicate container name will be fixed after #1676 is resolved.
